openassessment/xblock/static/js/spec/lms/oa_response.js

```diff
@@ -583,6 +580,65 @@ describe("OpenAssessment.ResponseView", function() {
         expect(fileUploader.uploadArgs[1].data).toEqual(files[1]);
     });
 
+    it("tests that new file uploads are appended", function() {
+        view.textResponse = 'optional';
+        view.fileUploadResponse = 'required';
+
+        expect(view.submitEnabled()).toBe(false);
+
+        // Upload files
+        var files = [{type: 'image/jpeg', size: 1024, name: 'picture1.jpg', data: ''},
+                     {type: 'image/jpeg', size: 1024, name: 'picture2.jpg', data: ''}];
+        view.prepareUpload(files, 'image', ['i1', 'i2']);
+        view.uploadFiles()
+        view.checkSubmissionAbility(true);
+        expect(view.submitEnabled()).toBe(true);
+        expect(view.files.length).toEqual(2);
+        view.prepareUpload(files, 'image', ['i1', 'i2']);
+        view.uploadFiles()
+        view.checkSubmissionAbility(true);
+        expect(view.files.length).toEqual(2);
+    });
+
+    it("tests that file upload number can't exceed maximum", function() {
+        spyOn(view.baseView, 'toggleActionError').and.callThrough();
+        spyOn(view, 'getSavedFileCount').and.returnValue(20);
+        view.textResponse = 'optional';
+        view.fileUploadResponse = 'required';
+        expect(view.submitEnabled()).toBe(false);
+        // Upload files
+        var files = [{type: 'image/jpeg', size: 1024, name: 'picture1.jpg', data: ''},
```

Comment:
:) This can be created more efficiently with a for-loop.

Reply:
copy and paste got away from me :)